src/pages/index.js

```diff
@@ -23,18 +29,85 @@ import {
   profileDescriptionInput,
 } from "../utils/constants.js";
 
-/* ----------------------- */
-/*     Profile Edit        */
-/* ----------------------- */
-const profileEditModalFormValidator = new FormValidator(
-  editProfileForm,
-  options
-);
-const profileUserInfo = new UserInfo({
+
+/*-----------------*/
+/*       Api      */
+/*---------------*/
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: authorizationCode,
+  },
+});
+
+
+api.getInitialCards().then((cards) => {
+  cardSection.renderItems(cards);
+  })
+  .catch((err) => {console.error("Error fetching cards:", err);
+});
+
+api.getProfile().then((data) => {
+  console.log('Profile data fetched:', data);
+  profileUserInfo.setUserInfo(data.name, data.about);
+  profileUserInfo.setUserAvatar(data.avatar);
+})
+.catch((err) => console.error('Error fetching profile:', err));
+
+//pop up with form
+const profileUserInfo = new userInfo({
   profileTitleSelector: ".profile__title",
   profileDescriptionSelector: ".profile__description",
+  profileAvatarSelector: ".profile__image"
 });
 
+const addCardModal = new PopupWithForm( "#add-card-modal", handleAddCardFormSubmit);
+addCardModal.setEventListeners();
+
+const editProfileModal = new PopupWithForm("#profile-edit-modal", handleProfileEditSubmit);
+editProfileModal.setEventListeners();
+
+const createCard = (items) => {
+  const newCard = getCard(items);
+  cardSection.addItem(newCard);
+};
+
+const cardSection = new Section(
+  { items: [], renderer: createCard },
+  ".cards__list"
+);
+
+const editAvatarPopup = new PopupWithForm("#avatar-modal", handleAvatarSubmit);
+editAvatarPopup.setEventListeners();
+
+const previewImageModal = new PopupWithImage({
+  modalSelector:"#preview-image-modal",});
+previewImageModal.setEventListeners();
+
+const deleteSubmitConfirmModal = new PopupWithConfirmation("#delete-modal", handleCardDeleteSubmit);
+deleteSubmitConfirmModal.setEventListeners();
+
+
+/* ------------------- */
+/*     Validation     */
+/* ----------------- */
+
+const addCardFormValidator = new FormValidator(addCardEditForm, options);
+
+const avatarFormValidator = new FormValidator(avatarEditForm, options);
+
+const profileEditModalFormValidator = new FormValidator(editProfileForm, options); 
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```